### PR TITLE
add unsigned char compilation for 2D images

### PR DESCRIPTION
**Pull request by NHPatterson**

### CODE REVIEW

File: .github/workflows/build-test-package.yml [hunk 1]
```diff
@@ -343,7 +343,7 @@ jobs:
       if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
       uses: pypa/gh-action-pypi-publish@master
       with:
-        user: __token__
+        user: nhpatterson
```
Comment (dzenanz) [inline .github/workflows/build-test-package.yml:346]: these changes should probably be reverted

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,6 +12,7 @@ if(ELASTIX_USE_OPENCL)
 endif()
 set(ELASTIX_BUILD_EXECUTABLE OFF)
 set(ELASTIX_BUILD_EXECUTABLE OFF CACHE BOOL "Generate executable or library")
+set( ELASTIX_IMAGE_2D_PIXELTYPES "unsigned char;float" CACHE STRING "Specify 2D pixel types" )
```
Comment (dzenanz) [inline CMakeLists.txt:15]: style consistency: no space between parentheses and their content
```suggestion
set(ELASTIX_IMAGE_2D_PIXELTYPES "unsigned char;float" CACHE STRING "Specify 2D pixel types")
```

File: setup.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
     author_email='itk+community@discourse.itk.org',
     packages=['itk'],
     package_dir={'itk': 'itk'},
-    download_url=r'https://github.com/InsightSoftwareConsortium/ITKElastix',
+    download_url=r'https://github.com/NHPatterson/ITKElastix',
```
Comment (dzenanz) [inline setup.py:26]: Did you intend to change this?

**Review by dzenanz — COMMENTED** [review]
I am not sure whether a closed PR can be re-opened, or a new one must be created.

**Comment by NHPatterson** [comment]
Apologies, I opened this PR from fork on this repo on accident, I mean to open the PR on my fork. I don't know the CMake and build process too well so I'm experimenting and thought to deploy these changes into artifacts through GH actions and then test and make sure they work. At first pass, ElastixRegistrationMethod doesn't accept `UC` images despite the compilation with the correct variable... this may just be my lack of experience (I don't know how to build the python wheels locally besides through CI). Any suggestions would be appreciated.

**Comment by dzenanz** [comment]
`ElastixRegistrationMethod` only supports `float` images. It is not trivial to lift that restriction, mostly due to `elastix` internals

**Comment by NHPatterson** [comment]
I'm confused a bit.. I believe that this setting `set( ELASTIX_IMAGE_2D_PIXELTYPES "unsigned char;float" CACHE STRING "Specify 2D pixel types" )` will compile all elastix components to support `UC` pixel type (@N-Dekker ?).  I did generate working [python wheels](https://github.com/NHPatterson/ITKElastix/actions/runs/1523037341) with this setting with this but got the error which may be downstream in ITK. It's not clear. 

### set up images
```python
import numpy as np
import itk

np_im = np.random.randint(0, 255, (1024, 1024), dtype=np.uint8)
itk_im = itk.GetImageFromArray(np_im)
itk_im
```
### 2D UC image in memory
```
<itk.itkImagePython.itkImageUC2; proxy of <Swig Object of type 'itkImageUC2 *' at 0x000001C34D93B840> >
```

### elastix registration instantiation
```python
elx_reg = itk.ElastixRegistrationMethod.New(itk_im, itk_im)
```
### trace
```python
Traceback (most recent call last):
  File "C:\miniconda3\envs\itkelx-wsi\lib\site-packages\IPython\core\interactiveshell.py", line 3457, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-10-1e6859a922f1>", line 1, in <module>
    elx_reg = itk.ElastixRegistrationMethod.New(itk_im, itk_im)
  File "C:\miniconda3\envs\itkelx-wsi\lib\site-packages\itk\support\template_class.py", line 733, in New
    raise itk.TemplateTypeError(self, input_type)
itk.support.extras.TemplateTypeError: itk.ElastixRegistrationMethod is not wrapped for input type `None`.
To limit the size of the package, only a limited number of
types are available in ITK Python. To print the supported
types, run the following command in your python environment:
    itk.ElastixRegistrationMethod.GetTypes()
Possible solutions:
* If you are an application user:
** Convert your input image into a supported format (see below).
** Contact developer to report the issue.
* If you are an application developer, force input images to be
loaded in a supported pixel type.
    e.g.: instance = itk.ElastixRegistrationMethod[itk.Image[itk.F,2], itk.Image[itk.F,2]].New(my_input)
* (Advanced) If you are an application developer, build ITK Python yourself and
turned to `ON` the corresponding CMake option to wrap the pixel type or image
dimension you need. When configuring ITK with CMake, you can set
`ITK_WRAP_${type}` (replace ${type} with appropriate pixel type such as
`double`). If you need to support images with 4 or 5 dimensions, you can add
these dimensions to the list of dimensions in the CMake variable
`ITK_WRAP_IMAGE_DIMS`.
Supported input types:
itk.Image[itk.F,2]
itk.Image[itk.F,3]
itk.Image[itk.F,4]
itk.Image[itk.D,2]
itk.Image[itk.D,3]
itk.Image[itk.D,4]

```

### elastix types
```python
itk.ElastixRegistrationMethod.GetTypes()
<itkTemplate itk::ElastixRegistrationMethod>
Options:
  [<class 'itk.itkImagePython.itkImageD2'>, <class 'itk.itkImagePython.itkImageD2'>]
  [<class 'itk.itkImagePython.itkImageD3'>, <class 'itk.itkImagePython.itkImageD3'>]
  [<class 'itk.itkImagePython.itkImageD4'>, <class 'itk.itkImagePython.itkImageD4'>]
  [<class 'itk.itkImagePython.itkImageF2'>, <class 'itk.itkImagePython.itkImageF2'>]
  [<class 'itk.itkImagePython.itkImageF3'>, <class 'itk.itkImagePython.itkImageF3'>]
  [<class 'itk.itkImagePython.itkImageF4'>, <class 'itk.itkImagePython.itkImageF4'>]
```

### confusing part in trace
```
itk.support.extras.TemplateTypeError: itk.ElastixRegistrationMethod is not wrapped for input type `None`.
To limit the size of the package, only a limited number of
types are available in ITK Python.
```

**Comment by NHPatterson** [comment]
Further investigation, it seems the ITKPythonPackage is only set up to wrap a [few pixel types](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/blob/8f5fe07ed55994923ebf21b7ff5b9dd800ca6a31/scripts/windows_build_wheels.py#L94-L97).

**Comment by dzenanz** [comment]
Those types are in addition to [the defaults](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Wrapping/WrappingOptions.cmake), e.g. https://github.com/InsightSoftwareConsortium/ITK/blob/047fa6934ac32711d1d7b84fa6eb81c72db01960/Wrapping/WrappingOptions.cmake#L23-L25.

**Pull request closed without merging**